python/cuml/common/input_utils.py

```diff
@@ -533,7 +533,7 @@ def input_to_host_array(X,
 
     if isinstance(X, (int, float, complex, bool, str,
                       type(None), dict, set, list, tuple)):
-        return X
+        return (X,)
```

Comment:
This does not seem like the desired behavior for list and tuple at minimum. If this function isn't giving you what you need for this PR, we might consider merging #4908 first and then building on it.

Reply:
The only goal here is to return the value unmodified. `input_to_host_array` currently returns a tuple. This allows the value (either an array or some other type) to always be set as the first element. Yes, merging this PR would allow not use a named tuple anymore if I understood correctly.

Reply:
Ah, I understand. Yes, this seems like a perfectly satisfactory workaround for the current named tuple setup.